test/controllers/transfer/ChooseYearsControllerTest.scala

```diff
@@ -0,0 +1,199 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.transfer
+
+import controllers.ControllerViewTestHelper
+import controllers.actions.AuthRetrievals
+import controllers.auth.PertaxAuthAction
+import helpers.FakePertaxAuthAction
+import models.{ApplyForEligibleYears, CurrentAndPreviousYearsEligibility, TaxYear}
+import org.mockito.ArgumentMatchers
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito._
+import org.scalatest.matchers.must.Matchers.convertToAnyMustWrapper
+import play.api.Application
+import play.api.i18n.MessagesApi
+import play.api.inject.bind
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.test.Helpers._
+import play.api.test.{FakeRequest, Injecting}
+import services.CacheService.CACHE_CHOOSE_YEARS
+import services.{CachingService, TimeService, TransferService}
+import test_utils.data.RecipientRecordData
+import uk.gov.hmrc.time
+import utils.{ControllerBaseTest, MockAuthenticatedAction, TransferErrorHandler}
+import views.html.errors.recipient_not_found
+
+import java.time.LocalDate
+import scala.concurrent.Future
+
+class ChooseYearsControllerTest extends ControllerBaseTest with ControllerViewTestHelper with Injecting {
+
+  val currentTaxYear: Int = time.TaxYear.current.startYear
+  val mockTransferService: TransferService = mock[TransferService]
+  val mockCachingService: CachingService = mock[CachingService]
+  val mockErrorHandler: TransferErrorHandler = mock[TransferErrorHandler]
+  val mockTimeService: TimeService = mock[TimeService]
+
+  override def fakeApplication(): Application = GuiceApplicationBuilder()
+    .overrides(
+      bind[TransferService].toInstance(mockTransferService),
+      bind[CachingService].toInstance(mockCachingService),
+      bind[TransferErrorHandler].toInstance(mockErrorHandler),
+      bind[TimeService].toInstance(mockTimeService),
+      bind[AuthRetrievals].to[MockAuthenticatedAction],
+      bind[MessagesApi].toInstance(stubMessagesApi()),
+      bind[PertaxAuthAction].to[FakePertaxAuthAction]
+    )
+    .build()
+
+  def controller: ChooseYearsController =
+    app.injector.instanceOf[ChooseYearsController]
+
+  when(mockTimeService.getCurrentDate) thenReturn LocalDate.now()
+  when(mockTimeService.getCurrentTaxYear) thenReturn currentTaxYear
+  when(mockTimeService.getStartDateForTaxYear(any())).thenReturn(time.TaxYear.current.starts)
+  when(mockTransferService.getCurrentAndPreviousYearsEligibility(any(), any())).thenReturn(
+    Future.successful(
+      CurrentAndPreviousYearsEligibility(
+        currentYearAvailable = true,
+        List(TaxYear(2015)),
+        RecipientRecordData.recipientRecord.data,
+        RecipientRecordData.recipientRecord.availableTaxYears
+      )
+    )
+  )
```

Comment:
You probably want all of this stubbing inside the beforeEach and to reset any mocks you're stubbing. Also do `.thenReturn(...)` rather than the infix `thenReturn` as it'll complain when it gets a Scala 3 upgrade.